### PR TITLE
scalar_scanner.rb - delete is typically 4 to 5 times faster than gsub

### PR DESCRIPTION
**Pull request by MSP-Greg**
Ok, locally on Windows, 4 to 5 times faster, with WSL2/Ubuntu, 3 times.

### CODE REVIEW

**Comment by MSP-Greg** [comment]
Re benchmarks, I used:
```ruby
require 'benchmark'

n = 500_000

INT = %w[123,456 123_456 123,_456_ 123,_456, 123,_456_, 123_,456_, ]

FLOAT = %w[123,456 123,456.001_02 123_456 123_456.001_02]

puts "#{'-' * 50} Integer Test"

Benchmark.bmbm do |x|
  x.report("gsub") {
    n.times { INT.each { |s| s.gsub(/,_/, '') } }
  }
  x.report("tr") {
    n.times { INT.each { |s| s.tr ',_', '' } }
  }
end

puts "\n#{'-' * 50} Float Test"

Benchmark.bmbm do |x|
  x.report("gsub") {
    n.times { FLOAT.each { |s| s.gsub(/[,_]|\.([Ee]|$)/, '\1') } }
  }
  x.report("tr") {
    n.times { FLOAT.each { |s| s.tr(',_', '').gsub(/\.([Ee]|$)/, '\1') } }
  }
end
```

File: lib/psych/scalar_scanner.rb [hunk 1]
```diff
@@ -88,7 +88,7 @@ def tokenize string
         if string.match?(/\A[-+]?\.\Z/)
           string
         else
-          Float(string.gsub(/[,_]|\.([Ee]|$)/, '\1'))
+          Float(string.tr(',_', '').gsub(/\.([Ee]|$)/, '\1'))
```
[the remark this answers is not in the document]
Comment (MSP-Greg) [reply lib/psych/scalar_scanner.rb:91]: @nobu  Thank you.  I'm sure I've seen the docs for `#delete`, but...

**Comment by MSP-Greg** [comment]
@tenderlove

Thanks.  JFYI, in the Actions Windows workflow (windows.yml), MSP-Greg/setup-ruby-pkgs is used.  It is now ruby/setup-ruby-pkgs.

But, for Windows jobs using windows-latest (windows-2022) image, all the dependencies for mingw/ucrt builds are installed by ruby/setup-ruby.  Hopefully, the same will be done soon for mswin builds, so it can then be changed to ruby/setup-ruby...

**Comment by tenderlove** [comment]
@MSP-Greg cool, thanks for the information! Also thanks for the PR 🙇🏻‍♀️

**Pull request merged**